core/src/main/scala/org/apache/spark/executor/ExecutorMetricsPoller.scala

```diff
@@ -147,6 +147,10 @@ private[spark] class ExecutorMetricsPoller(
    * Called by TaskRunner#run.
    */
   def getTaskMetricPeaks(taskId: Long): Array[Long] = {
+    // If task running duration is less then pollInterval, task metric peak may not updated
+    // call poll() update to latest metrics
+    poll()
+
```

Comment:
@jiangxb1987  poll() at here when task finish or error will get latest metrics peak.

ExecutorMtericPoller.onTaskComplet() just remove taskId corresponding  entry in Map